src/plugins/discover/public/application/main/utils/get_switch_data_view_app_state.ts

```diff
@@ -37,15 +37,17 @@ export function getDataViewAppState(
     );
   }
 
-  if (query && isOfAggregateQueryType(query)) {
+  const isTextBasedQueryMode = !!query && isOfAggregateQueryType(query);
```

Comment:
I think you could omit `!!query`, since it is checked in isOfAggregateQueryType
```suggestion
  const isTextBasedQueryMode = isOfAggregateQueryType(query);
```